src/scripts/main.js

```diff
@@ -1,11 +1,30 @@
 'use strict';
 
+const body = document.querySelector('body');
+
 function waitFor(element, eventName) {
-  // write your code here
+  const error = `It was NOT ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`;
+  const promise = new Promise((resolve, reject) => {
+    if (!element) {
+      reject(error);
```

Comment:
The `waitFor` function immediately rejects the promise if the element is not found. Consider adding error handling or logging to inform the user about the missing element.